src/rest/handlers/rest_queries.erl

```diff
@@ -101,28 +101,95 @@ execute(Fun, Args, FieldName) ->
 
 %% @todo
 -spec reaches_naive(binary(), binary()) -> {ok, boolean()} | {error, any()}.
-reaches_naive(_From, _To) ->
-    {ok, false}.
+reaches_naive(From, To) ->
+    case do_members_naive(To) of
+        {error, Error} -> {error, Error};
+        {ok, Members} -> {ok, #{<<"reaches">> => sets:is_element(From, Members), <<"duration">> => 0}}
+    end.
 
 %% @todo
 -spec reaches_indexed(binary(), binary()) -> {ok, boolean()} | {error, any()}.
 reaches_indexed(_From, _To) ->
     {ok, false}.
 
 %% @todo
--spec effective_permissions_naive(binary(), binary()) -> {ok, binary()} | {error, any()}.
-effective_permissions_naive(_From, _To) ->
-    {ok, <<"">>}.
+-spec effective_permissions_naive(binary(), binary()) -> {ok, map()} | {error, any()}.
+effective_permissions_naive(From, To) ->
+    MyZoneId = gmm_utils:zone_id(),
+    case gmm_utils:owner_of(From) of
+        MyZoneId -> {ok, #{<<"effectivePermissions">> => do_effective_permissions_naive(From, To), <<"duration">> => 0}};
+        Other -> zone_client:effective_permissions(naive, Other, From, To)
+    end.
+
+do_effective_permissions_naive(From, To) ->
+    JoinPermissions = fun
+        (<<"no-route">>, Permissions) -> Permissions;
+        (Permissions, <<"no-route">>) -> Permissions;
+        (A, B) -> gmm_utils:permissions_or(A,B)
+    end,
+    case graph:list_children(From) of
```

Comment:
Pomyliłeś kierunek krawędzi.

```suggestion
    case graph:list_children(To) of
```

Patrzymy na ostatnie krawędzie na ścieżkach od From do To. Gdzie From jest "dalekim dzieckiem" To